src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogViewModel.java

```diff
@@ -0,0 +1,324 @@
+                                                                                      ButtonType.OK, openHelp);
+
+            result.filter(btn -> btn.equals(openHelp)).ifPresent(btn -> HelpAction.openHelpPage(HelpFile.SQL_DATABASE_MIGRATION));
+            result.filter(btn -> btn == ButtonType.OK).ifPresent(btn -> openSharedDatabase());
+
+        }
+        loading.set(false);
+    }
+
+    private void setPreferences() {
+        prefs.setType(selectedDBMSType.getValue().toString());
+        prefs.setHost(host.getValue());
+        prefs.setPort(port.getValue());
+        prefs.setName(db.getValue());
+        prefs.setUser(user.getValue());
+
+        if (rememberPassword.get()) {
+            try {
+                prefs.setPassword(new Password(password.getValue(), password.getValue()).encrypt());
+            } catch (GeneralSecurityException | UnsupportedEncodingException e) {
+                LOGGER.error("Could not store the password due to encryption problems.", e);
+            }
+        } else {
+            prefs.clearPassword(); // for the case that the password is already set
+        }
+
+        prefs.setRememberPassword(rememberPassword.get());
+    }
+
+    /**
+     * Fetches possibly saved data and configures the control elements respectively.
+     */
+    public void applyPreferences() {
+        Optional<String> sharedDatabaseType = prefs.getType();
+        Optional<String> sharedDatabaseHost = prefs.getHost();
+        Optional<String> sharedDatabasePort = prefs.getPort();
+        Optional<String> sharedDatabaseName = prefs.getName();
+        Optional<String> sharedDatabaseUser = prefs.getUser();
+        Optional<String> sharedDatabasePassword = prefs.getPassword();
+        boolean sharedDatabaseRememberPassword = prefs.getRememberPassword();
+
+        if (sharedDatabaseType.isPresent()) {
+            Optional<DBMSType> dbmsType = DBMSType.fromString(sharedDatabaseType.get());
+            dbmsType.ifPresent(selectedDBMSType::set);
+        }
+
+        sharedDatabaseHost.ifPresent(host::set);
+        sharedDatabasePort.ifPresent(port::set);
+        sharedDatabaseName.ifPresent(db::set);
+        sharedDatabaseUser.ifPresent(user::set);
+
+        if (sharedDatabasePassword.isPresent() && sharedDatabaseUser.isPresent()) {
+            try {
+                password.setValue(new Password(sharedDatabasePassword.get().toCharArray(), sharedDatabaseUser.get()).decrypt());
+            } catch (GeneralSecurityException | UnsupportedEncodingException e) {
+                LOGGER.error("Could not read the password due to decryption problems.", e);
+            }
+        }
+
+        rememberPassword.set(sharedDatabaseRememberPassword);
+    }
+
+    private boolean isSharedDatabaseAlreadyPresent() {
+        List<BasePanel> panels = frame.getBasePanelList();
+        return panels.parallelStream().anyMatch(panel -> {
```

Comment:
Can you try to move this check using `frame` to `StateManager` (it also gives access to the list of open databases).

Reply:
It currently only support the active database, needs probably be extended in JabRefFrame, there is this line:   EasyBind.subscribe(tabbedPane.getSelectionModel().selectedItemProperty(), e -> {